applications/ConstitutiveModelsApplication/custom_models/constitutive_model_data.hpp

```diff
@@ -0,0 +1,692 @@
+    {
+      StressMeasure_PK1,            //stress related to reference configuration non-symmetric
+      StressMeasure_PK2,            //stress related to reference configuration
+      StressMeasure_Kirchhoff,      //stress related to current   configuration
+      StressMeasure_Cauchy          //stress related to current   configuration
+    };
+
+
+    struct MaterialData
+    {      
+    public:
+      
+      //general elastic material properties 
+      double PoissonCoefficient;
+      double YoungModulus;
+      double LameMu;
+      double LameMuBar;
+      double LameLambda;
+      double BulkModulus;
+
+      //general hyperelastic material properties
+      std::vector<double> ModelParameters;
+
+    public:
+      
+      //Get const Data
+      const double& GetPoissonCoefficient() const {return PoissonCoefficient;};
+      const double& GetYoungModulus      () const {return YoungModulus;};
+      const double& GetLameMu            () const {return LameMu;};
+      const double& GetLameMuBar         () const {return LameMuBar;};   
+      const double& GetLameLambda        () const {return LameLambda;};
+      const double& GetBulkModulus       () const {return BulkModulus;};
+      
+      const std::vector<double>& GetModelParameters   () const {return ModelParameters;};
+
+    };
+
+
+    template <class T>
+    struct VariableValue
+    {
+    private:
+      
+      const Variable<T> *mpVariable;
+      T *mpValue;
+
+    public:
+
+      //constructors
+      VariableValue(){};
+      VariableValue(const Variable<T>& rVariable, T& rValue){mpVariable = &rVariable; mpValue = &rValue;};
+      
+      //Set Data Pointers
+      void SetVariableValue            (const Variable<T>& rVariable, T& rValue) {mpVariable = &rVariable; mpValue = &rValue;};
+      void SetVariable                 (const Variable<T>& rVariable) {mpVariable = &rVariable;};
+      void SetValue                    (T& rValue) {mpValue = &rValue;};
+
+      //Get Data Pointers
+      
+      //Acces non const Data
+      T&                 rValue        (){return *mpValue;};
+      
+      //Get const Data	
+      const Variable<T>& GetVariable   () const {return *mpVariable;};
+      const T&           GetValue      () const {return *mpValue;};
+    };
+    
+
+    struct VariableValueData
+    {
+    private:
+
+      enum VarType { INTEGER, DOUBLE, VECTOR, MATRIX, ARRAY3, ARRAY6 };
+
+      VarType                                        mType;      
+      VariableValue<int>                    *mpIntVariable;
+      VariableValue<double>              *mpDoubleVariable;
+      VariableValue<Vector>              *mpVectorVariable;
+      VariableValue<Matrix>              *mpMatrixVariable;
+      VariableValue<array_1d<double,3> > *mpArray3Variable;
+      VariableValue<array_1d<double,6> > *mpArray6Variable;
+      
+   public:
+
+      //Constructor
+      VariableValueData()
+      {
+	mpIntVariable    = NULL;
```

Comment:
I recommend to use nullptr https://stackoverflow.com/questions/20509734/null-vs-nullptr-why-was-it-replaced

Can be replaced using clang tidy https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-nullptr.html

https://github.com/KratosMultiphysics/Kratos/wiki/How-to-use-Clang-Tidy-to-automatically-correct-code